Core.Collectors/Collector/CachingCollector.cs

```diff
@@ -25,7 +26,12 @@ public CachingCollector(CollectorBase<TCollectionNode> collector, ITelemetryClie
             this.exceptions = new List<Exception>();
         }
 
-        public async Task<bool> ProcessAndCacheAsync(TCollectionNode collectionNode, TEndpointProgressTableEntity progressRecord, bool ignoreCache, bool scheduledCollection)
+        public Task<bool> ProcessAndCacheAsync(TCollectionNode collectionNode, TEndpointProgressTableEntity progressRecord, bool ignoreCache, bool scheduledCollection)
```

Comment:
If this method is not async then why does its name states async?

Reply:
Anything that returns a Task is essentially an async method really (since on the caller end you either need to await on it or you need to use fire-and-forget).